ini/INIReader.h

```diff
@@ -380,9 +387,23 @@ inline std::vector<T> INIReader::GetVector(std::string section, std::string name
     if (!_values.count(key)) {
         return default_v;
     }
-    return GetVector<T>(section, name);    
+    return GetVector<T>(section, name);
 }
 
+inline const INIReader::_Get(std::string section) const {
+    if (!_values.count(section)) {
+        throw std::runtime_error("section " + section + " not found.");
+    }
+    return _values.at(key);
+}
+
+inline const INIReader::_Get(std::string section, std::string name) const {
+    std::map<std::string, std::string> _section = _Get(section)
+    if (!_section.count(name)) {
+        return "";

```

Comment:
why not throw an exception?

Reply:
Because `.Get()` with a default value needs to catch the exception this way. I read that catching exception is expensive in C++, and I didn't see a "key not found" exception so it risks catching unexpected errors.

Reply:
suggest implementataion, let `std::at()` handle access value or throw `out_of_range` error. ref: http://www.cplusplus.com/reference/map/map/at/

```cpp
inline const INIReader::_Get(std::string section, std::string name) const {
    std::map<std::string, std::string> _section = _Get(section).at(name
}
```

in my opinion, as a library, we need to throw error explicitly, not silently ignore or return special behavior (like returning an empty string), it may confuse ppl.

Reply:
after read more carefully, i now understand you want `_Get` as an internal helper function, I would suggest you implement like this:

```cpp
template<typename T = std::string>
inline T INIReader::Get(std::string section, std::string name) const {
    try{
        std::string value = _values.at(section).at(name);
    } catch(std::out_of_range &e) { 
        // only handle key not found error, others err with throw directly.
        throw std::runtime_error("key " + section + name + " not found.");
    }
    return Parser<T>(value);
}
```
```

Reply:
This doesn't work because `value` is in `try`'s scope.